silence/silence.go

```diff
@@ -55,53 +62,64 @@ func FindExisting() (string, error) {
 		err = json.Unmarshal(resp, &silences)
 		if err != nil {
 			log.Printf("There was an error unmarshalling get silence response")
-			return "", err
+			return sr, err
 		}
+
 		if len(silences) == 0 {
 			log.Printf("No Silences Present")
-			return "", nil
+			return sr, nil
 		}
 
 		for _, silence := range silences {
 			if silence.CreatedBy != createdBy {
 				continue
 			}
-			if silence.Status.State != "active" {
+
+			if !silence.Active() {
 				log.Printf("Silence is not active.")
 				continue
 			}
-			log.Printf("Found silence created by job: %s", silence.ID)
-			return silence.ID, nil
+
+			sr = silence
+			log.Printf("Found silence created by job: %s", sr.ID)
+
+			return sr, nil
 		}
 
 		log.Printf("No silences created by job found.")
-		return "", nil
+		return sr, nil
 	}
 
-	return "", fmt.Errorf("unable to get a list of existing silences")
+	return sr, fmt.Errorf("unable to get a list of existing silences")
 }
 
-// Create adds a new silence that expires in one hour.
-func Create() (string, error) {
+func (sr *SilenceRequest) Build(expiryPeriod time.Duration) *SilenceRequest {
 	// Create the Silence
 	now := time.Now().UTC()
-	end := now.Add(1 * time.Hour)
+	end := now.Add(expiryPeriod)
 
 	allMatcher := matcher{}
 	allMatcher.Name = "severity"
-	allMatcher.Value = "info|warning|critical"
+	// Match all alerts
+	// This doesn't work and should, maybe it needs better escaping?
+	// allMatcher.Value = ".*"
```

Comment:
We can't match _all_ alerts because of the Watchdog alert, if we also match and silence that then we'll get a DMS alert for each cluster.

Is there a reason we're not silencing `info` alerts, and are we sure that the capital letters will match?  I didn't test with capitals before but IIRC they were coming in at the time all lowercase.

Reply:
Ah, you are correct I'll update this to what we had `info|warning|critical` good pickup.